velox/exec/OutputBuffer.h

```diff
@@ -74,6 +74,41 @@ class ArbitraryBuffer {
   std::deque<std::shared_ptr<SerializedPage>> pages_;
 };
 
+struct BufferInfo {
+  void recordAddPage(const SerializedPage& data) {
+    VELOX_CHECK(data.rows() >= 0, "SerializedPage's row size is not inited");
+    int64_t rows = data.rows();
+    int64_t size = data.size();
+    bufferedBytes += size;
+    bufferedRows += rows;
+    bufferedPages++;
+    bytesAdded += size;
+    rowsAdded += rows;
+    pagesAdded++;
+  }
+  void recordSendPage(const SerializedPage& data) {
+    VELOX_CHECK(data.rows() >= 0, "SerializedPage's row size is not inited");
+    int64_t rows = data.rows();
+    int64_t size = data.size();
+    bufferedBytes -= size;
+    bufferedRows -= rows;
+    bufferedPages--;
+    bytesSent += size;
+    rowsSent += rows;
+    pagesSent++;
+  }
+  bool finished{false};
```

Comment:
add empty line

Add comments to each field.

Reply:
done.